draft-ietf-rtcweb-jsep.xml

```diff
@@ -1626,36 +1626,53 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
         </section>
         <section title="addIceCandidate" anchor="sec.addicecandidate">
 
-          <t>The addIceCandidate method provides a remote candidate to
-          the ICE agent, which, if parsed successfully, will be added
+          <t>The addIceCandidate method provides an update to
+          the ICE agent via an IceCandidate object 
+          <xref target="sec.ice-candidate-format"/>. If the
+          IceCandidate's candidate field is filled in, the IceCandidate
+          is treated as a new remote ICE candidate, which will be added
           to the current and/or pending remote description according to
-          the rules defined for Trickle ICE. The pair of MID and ufrag
-          is used to determine the m= section and ICE candidate
-          generation to which the candidate belongs. If the MID is not
-          present, the m= section index is used to look up the locally
-          generated MID (see
-          <xref target="sec.applying-a-remote-desc" />), which is used
-          in place of a supplied MID. If these values or the candidate
-          string are invalid, an error is generated.</t>
-
-          <t>The purpose of the ufrag is to resolve ambiguities when
-          trickle ICE is in progress during an ICE restart. If the
-          ufrag is absent, the candidate MUST be assumed to belong to
-          the most recently applied remote description. Connectivity
-          checks will be sent to the new candidate.</t>
-
-          <t>This method can also be used to provide an
-          end-of-candidates indication to the ICE agent, as defined in
-          <xref target="I-D.ietf-ice-trickle" />). The MID and ufrag
-          are used as described above to determine the m= section and
-          ICE generation for which candidate gathering is complete. If
-          the ufrag is not present, then the end-of-candidates
-          indication MUST be assumed to apply to the relevant m=
-          section in the most recently applied remote description. If
-          neither the MID nor the m= index is present, then the
+          the rules defined for Trickle ICE. Otherwise, the IceCandidate
+          is treated as an end-of-candidates indication, as defined in
+          <xref target="I-D.ietf-ice-trickle" />).</t>
+          
+          <t>In either case, the MID and ufrag fields from the supplied 
```

Comment:
I think this sentence needs some tweaking to take care of case there is an m= index but no MID